TSC glib patches

dchengTSC opened the pull request:
Changes to the glib portfile:
1. glib can now be successfully cross-compiled on mac from x86_64 to arm64 and vice versa
2. `g_thread_new` won't fail if unable to inherit the current thread's priority

dchengTSC committed:
Fix cross-compiling glib

Copied patch from https://github.com/microsoft/vcpkg/pull/26259#issuecomment-1518395642

dchengTSC committed:
Allow setting thread priority to fail on Windows

mike-malburg reviewed (commented):
@dchengTSC Do we need to update the version in https://github.com/TechSmith/vcpkg/blob/master/ports/glib/vcpkg.json?

I noticed this error: https://github.com/TechSmith/vcpkg/actions/runs/5557235477/jobs/10150755906

> When making any changes to a library, the version or port-version in vcpkg.json must be modified, and the version database updated.

dchengTSC commented:
Yeah I don't know what that means. I assume that's only really really necessary if we want to merge to `Microsoft:master`.

I can dig into it some more, but I'm not too concerned about this.

mike-malburg reviewed (approved):
> Yeah I don't know what that means. I assume that's only really really necessary if we want to merge to Microsoft:master.
> 
> I can dig into it some more, but I'm not too concerned about this.

Changes seem fine to me.  That was my only reason for pause on this.

dchengTSC committed:
Update port-version to make vcpkg pr checker happy

dchengTSC committed:
More fixes based on PR check

Ran the following based on the PR checker error message:

git add -u && git commit
git checkout 291ed0c0880b3729ff9d11aee7c1f764a1ccb591 -- versions
./vcpkg x-add-version --all
